src/components/steps/nutrition/nutrition.tsx

```diff
@@ -1,42 +1,70 @@
-import { Fragment, useState } from 'react';
+import { Fragment, useContext, useEffect, useMemo, useState } from 'react';
+
+import { StorageContext } from 'context/storage.context';
 
 import { Container, TextField, Typography } from '@mui/material';
 
-type NutritionType = {
-  calories: string;
-  carbohydrates: string;
-  protein: string;
-  saturates: string;
-  sodium: string;
-  sugars: string;
-  totalFat: string;
-};
+import { Nutrition as NutritionType } from 'utils/api/api.types';
 
-const defaultNutrition = {
-  calories: '',
-  carbohydrates: '',
-  protein: '',
-  saturates: '',
-  sodium: '',
-  sugars: '',
-  totalFat: '',
-};
+import { StepsProps } from '../nameAndDescription/nameAndDescription';
+
+const Nutrition: React.FC<StepsProps> = ({ submitStep, setSubmitStep }) => {
+  const [nutrition, setNutrition] = useState<NutritionType>();
+
+  const { setRecipeToUpload, recipeToUpload } = useContext(StorageContext);
+
+  const registeredNutrition = useMemo(() => {
+    let defaultNutrition: NutritionType = {
```

Comment:
This entire function could be simplified. You could have something like this:
return recipeToUpload?.nutrition ?? { calories: undefined, carbohydrates: undefined .... }